src/H5MM.c

```diff
@@ -395,35 +147,12 @@ H5MM_realloc(void *mem, size_t size)
         /* Not defined in the standard, return NULL */
```

Comment:
From reading the standard it looks like NULL != mem && 0 == size is technically undefined. At least it is if malloc(0) is undefined.